beacon-chain/utils/shuffle.go

```diff
@@ -70,3 +79,80 @@ func SplitIndices(l []uint64, n uint64) [][]uint64 {
 	}
 	return divided
 }
+
+// PermutedIndex Return `p(index)` in a pseudorandom permutation `p` of `0...list_size - 1` with ``seed`` as entropy.
+//     Utilizes 'swap or not' shuffling
+//
+// Spec pseudocode definition:
+// def get_permuted_index(index: int, list_size: int, seed: Bytes32) -> int:
+//     """
+//     Return `p(index)` in a pseudorandom permutation `p` of `0...list_size - 1` with ``seed`` as entropy.
+//     Utilizes 'swap or not' shuffling found in
+//     https://link.springer.com/content/pdf/10.1007%2F978-3-642-32009-5_1.pdf
+//     See the 'generalized domain' algorithm on page 3.
+//     """
+//     assert index < list_size
+//     assert list_size <= 2**40
+//     for round in range(SHUFFLE_ROUND_COUNT):
+//         pivot = bytes_to_int(hash(seed + int_to_bytes1(round))[0:8]) % list_size
+//         flip = (pivot - index) % list_size
+//         position = max(index, flip)
+//         source = hash(seed + int_to_bytes1(round) + int_to_bytes4(position // 256))
+//         byte = source[(position % 256) // 8]
+//         bit = (byte >> (position % 8)) % 2
+//         index = flip if bit else index
+//     return index
+func PermutedIndex(index uint64, listSize uint64, seed [32]byte) (uint64, error) {
+	if params.BeaconConfig().ShuffleRoundCount == 0 {
+		return index, nil
+	}
+	// spec: assert index < list_size
+	if index >= listSize {
+		return 0, fmt.Errorf("input index %d out of bounds: %d",
+			index, listSize)
+	}
+	// spec: assert list_size <= 2**40
+	if listSize > uint64(math.Pow(2, 40)) {
```

Comment:
Please make this a constant, `maxShuffleListSize`

Reply:
done